sympy/core/numbers.py

```diff
@@ -440,9 +440,9 @@ def igcdex(a, b):
 
 def mod_inverse(a, m):
     """
-    Return the number c such that, ( a * c ) % m == 1 where
-    c has the same sign as a. If no such value exists, a
-    ValueError is raised.
+    Return the number c such that, ( a * c ) % m == ( 1 ) % m
```

Comment:
Why `( 1 ) % m` instead of `1`?

Reply:
Actually, its a congruence (as I wrote yesterday on gitter).
So `( 1 ) % m` denotes `1 modulo m` with same sign as `m`.

Reply:
To write mathematically, it just denotes the following piecewise define function:
```
( 1 ) % m = 1, for m>0
( 1 ) % m = m + 1, for m<0
```

Reply:
Python's `%` operator seems to work that way but I am not sure that `mod_inverse` in SymPy should necessarily copy its behaviour.

Reply:
Yes, WA also uses the same convention. Actually its more convenient, if modulo defines the allowed range of numbers, maintaining consistency.

Reply:
I think it would be more understandable to write it as `(a*c - 1) % m == 0` (or `a*c - 1` is divisible by `m` to avoid Python's `%`).

Reply:
Yes, actuall this [convention](https://en.wikipedia.org/wiki/Modulo_operation) was proposed by D.E. Knuth based on floor division.

Yes, I agree, to improve readability, we can just strip operators from the expression and write it as a congruence.
`a*c \equiv 1 \pmod{m}`
`a*c = 1 (mod m)`